src/Models.py

```diff
@@ -0,0 +1,661 @@
+        return self.redis_session.hget(self.model_id, b'folder_id')
+
+    def get_parent_folder_model(self):
+        return FolderModel(folder_id=self.get_parent_folder_id(), redis_session=self.redis_session)
+
+    def add_init_file_info(self, size, first_folder_id):
+        """
+        Saves to the DB the initialized file metadata: size of the file, the id for the files calls out set,
+        the id for the files calls in set, folder id, contained functions set id,and contained address.
+        Add the file id to the set of file ids.
+        Add the first function into the set of contained functions in the file.
+        """
+        self.basic_init_save(size=size)
+        self.redis_session.hset(self.model_id, b'contained_functions_set_id', self.contained_nodes_set_id)
+        self.redis_session.hset(self.model_id, b'folder_id', first_folder_id)
+        Files(self.redis_session).add_model_id(self.model_id)
+
+    def recursion_init(self, size):
+        """
+        Init the files metadata and initialize folders nodes.
+        """
+        first_folder_id = b'folder:' + self.contained_function_address
+        self.add_init_file_info(size, first_folder_id)
+        folder_model = FolderModel(folder_id=first_folder_id, redis_session=self.redis_session)
+        folder_model.recursion_init(size)
+        self.add_file_id_to_folder_contained_files(folder_model=folder_model)
+
+    def add_file_id_to_folder_contained_files(self, folder_model: FolderModel):
+        self.redis_session.sadd(folder_model.contained_nodes_set_id, self.model_id)
+
+    def recursion_add_edge(self, called_file_model):
+        """
+        Add edge to the called file and call add edge for folder
+        The edged contained in the files relations need to exist inside the folder relations
+        """
+        self.add_edge(called_file_model)
+        folder_model = FolderModel(folder_id=self.get_parent_folder_id(), redis_session=self.redis_session)
+        folder_model.add_folder_edge(called_file_model=called_file_model)
+
+    def remove(self):
+        self.redis_session.delete(self.calls_out_set_id)
+        self.redis_session.delete(self.calls_in_set_id)
+        self.redis_session.delete(self.contained_nodes_set_id)
+        Files(redis_session=self.redis_session).remove_model_id(self.model_id)
+        father_folder_model = FolderModel(folder_id=self.get_parent_folder_id(), redis_session=self.redis_session)
+        father_folder_model.remove_contained_file(self.model_id)
+        self.delete_model()
+
+    def recursion_cluster(self, file_to_cluster: FileModel):
+        current_folder_model = self.get_parent_folder_model()
+        folder_to_cluster = file_to_cluster.get_parent_folder_model()
+        self.cluster(file_to_cluster)
+        current_folder_model.cluster(folder_to_cluster)
+
+
+class FunctionModel(NodeModel):
+    __slots__ = 'model_id'
+
+    def __init__(self, redis_session: redis.Redis, address=None, function_id=None):
```

Comment:
Remove the usage of the address parameter. And use the function id similar to the other models such as file and folder.